### ISSUE TITLE
Do not add focus until the user has interacted with the table 

### ISSUE THREAD

**Issue by tgreenwatts** [palantir/blueprint#1098]
Related to #1072 
When focus cells are enbled, do not add focus until the user has interacted with the table 

Rationale:
When the table is being use as part of a larger application (i.e. less spreadsheet-like, more table-list) - the table does not focus until the user interacts 

Also partially fixes by removing the initial focus: https://github.com/palantir/blueprint/issues/1077

**Comment by cmslewis** [comment]
Hmm, maybe, but I always feel weird when it's impossible to get back to an initial state. In this case, the initial state is a focus-less table, and the first interaction reveals the focus-cell styling. How do you get back out of that? I think I'd want that to be possible.

**Comment by gscshoyru** [comment]
Agreed. If focus is enabled, we should start with a focus cell (and that cell selected) -- it should be possible to go back to initial states, and we should design things that way.

**Comment by llorca** [comment]
I suppose the focus cell could go away once you deselect everything in the table? Thoughts?

**Comment by gscshoyru** [comment]
If you have a focus cell you shouldn't be able to deselect all the things. The two are at odds.

**Comment by llorca** [comment]
Can you elaborate on what's odd about it? What if you have one cell that's focused (and therefore selected), and you `cmd+click` it? What do you see happening?

**Comment by gscshoyru** [comment]
I mean, it does in fact work that way presently, but with focus cell it doesn't make sense for such a thing to be possible. Having a focus cell should mean that something is always selected. Neither excel nor google sheets let you do this (though they do so in wildly different ways). Don't we already have tickets about turning off the fact that you can deselect like that?

**Comment by tgreenwatts** [comment]
I think should be be able to deselect and get back to the initial state - because this is a table. We can have something like `enforceFocus` if it's more spreadsheet-y

**Comment by tgreenwatts** [comment]
Updated thoughts: Let's wait and see how users are using the table and see if this is useful
